asdf/tags/core/__init__.py

```diff
@@ -41,25 +41,17 @@ class HistoryEntry(dict, AsdfType):
     version = '1.0.0'
 
 
-class ExtensionMetadata(AsdfType):
+class ExtensionMetadata(dict, AsdfType):
     name = 'core/extension_metadata'
     version = '1.0.0'
 
-    def __init__(self, extension_class=None, software={}):
-        self.extension_class = extension_class
-        self.software = software
+    @property
+    def extension_class(self):
+        return self['extension_class']
```

Comment:
If `extension_class` is not passed in the constructor, shouldn't it return `None`?  Something like

```
        return self.get("extension_class", None)
```

Reply:
`extension_class` is required by the schema:

https://asdf-standard.readthedocs.io/en/1.5.0/generated/stsci.edu/asdf/core/extension_metadata-1.0.0.html#Original%20Schema

so we would have hit a validation error before this point if the property was missing.

Reply:
Ah, ok.  No problem then.